src/main/java/org/moonlightcontroller/managers/ConnectionManager.java

```diff
@@ -106,6 +114,35 @@ public Response handleHelloRequest(Hello message) {
 		}
 	}
 
+	private List<JsonConnector> translateConnectors(List<IConnector> connectors) {
+		List<JsonConnector> jConnectors = new ArrayList<>();
+		for (IConnector connector : connectors) {
+			jConnectors.addAll(translateConnector(connector));
+		}
+
+		return jConnectors;
+	}
+
+	private List<JsonConnector> translateConnector(IConnector connector) {
+		List<Integer> dst_ports = connector.getDestBlock().getPorts();
+		List<JsonConnector> connectors = new ArrayList<>();
+		dst_ports.stream().forEach(dst_port -> connectors.add(new JsonConnector(connector.getSourceBlockId(), 

```

Comment:
Dst_port should be 0
The translation from IConnector to JsonConnector should be one to one, not one to many